### PR TITLE
fix(platform): approval flow - a11y issues

### LINKED ISSUE #5574: Accessibility:  Approval Flow-Platform Component
Accessibility: Approval Flow Component (There are couple of accessibility issues found while testing, so i have attached error document for you reference.)
Screenshot:
[Approval flow_Platform_ARIBA Controls.docx](https://github.com/SAP/fundamental-ngx/files/6654147/Approval.flow_Platform_ARIBA.Controls.docx)

### PR DESCRIPTION
**Pull request by dmitry-stepanenko**
## Related Issue(s)

closes #5574

## Description
- Issue 1 - done
- Issue 2 - cannot reproduce
- Issue 3 - done
- Issue 4 - done
- Issue 5 - done
- Issue 6 - done
- Issue 7 - cannot reproduce
- Issue 8 - done separately https://github.com/SAP/fundamental-ngx/pull/8160
- Issue 9 - not done, now scrollbar will always be there
- Issue 10 - not done, now scrollbar will always be there
- Issue 11 - cannot reproduce
- Issue 12 - cannot reproduce
- Issue 13 - done
- Issue 14 - cannot reproduce
- Issue 15 - done
- Issue 16 - done
- Issue 17 - done
- Issue 18 - done
- Issue 19 - cannot reproduce
- Issue 20 - done

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [n/a] different combinations of components - free style
-   [n/a] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [n/a] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [n/a] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [n/a] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

**Comment by Karthickc1994** [comment]
Hello Colleague,

The preview URL is not working please provide me the URL for RETEST.

Environment Used:
[https://deploy-preview-8164--fundamental-ngx.netlify.app/assets/typedoc/platform/classes/approvalflowcomponent]

Regards,
Karthikeyan.c

File: libs/platform/src/lib/approval-flow/approval-flow-node/approval-flow-node.component.html [hunk 1]
```diff
@@ -127,6 +135,7 @@
         *ngIf="node.status !== 'approved'"
         fd-button
         glyph="add"
+        title="Add node"
```
Comment (platon-rov) [inline libs/platform/src/lib/approval-flow/approval-flow-node/approval-flow-node.component.html:138]: please add i18n support.

File: libs/platform/src/lib/approval-flow/approval-flow-node/approval-flow-node.component.html [hunk 2]
```diff
@@ -27,7 +27,9 @@
         class="fdp-approval-flow-node__menu-button"
         fd-button
         fdType="transparent"
+        tabindex="0"
         glyph="overflow"
+        title="Menu"
```
Comment (platon-rov) [inline libs/platform/src/lib/approval-flow/approval-flow-node/approval-flow-node.component.html:32]: please add i18n support.

File: libs/core/src/lib/token/token.component.ts [hunk 3]
```diff
@@ -137,4 +145,10 @@ export class TokenComponent implements OnInit, AfterViewInit, OnDestroy {
     tokenClickHandler(event): void {
         this.onTokenClick.emit(event);
     }
+
+    /** @hidden */
+    setTotalCount(count: number): void {
```
Comment (platon-rov) [inline libs/core/src/lib/token/token.component.ts:150]: if hidden shouldn't this be _underlined?

**Comment by Karthickc1994** [comment]
Hello Colleague,

Please find the test results below:

Issue1: Fixed
Issue 2

Priority: High
Checkpoint: ACC-264

Navigate to the names available within watchers and activate it.
Observed Behavior:
When the dialog appears on the screen, the entire information available within the dialog is read at once.
The speech output is:
Modal dialog
Detail Melissa Martin Accounting team Contact info Mobile 1305237482483 Phone 1305237482483 E-mail [melissa.martin@company.com](mailto:melissa.martin@company.com) Company Name Company [Amelissa.martin@company.com](mailto:Amelissa.martin@company.com) Address 481 West Street, Anytown OH, 83749, USA
ContentInfoRegion
Cancel Button
To activate press
Enter.
Expected Behavior:
Ideally, when a dialog appears on the screen the entire information available should not be read at once instead the dialog header, along with the information regarding the focused element should be read.

Screenshot:

image


Issue3; Fixed

Issue 4

Priority: High
Checkpoint: ACC-264

Activate Marketing button in process. In Marketing dialog press tab to navigate to the list.
Observed Behavior:
On navigating o the list available within the dialog, the entire information available within the list is read at once.
Also, the item count for the number of items available within the list is not read.
The speech output is:
Modal dialog
Marketing team (2) Logan Herrera Sales team Caleb Taylor Marketing team
Logan Herrera Logan Herrera Sales team Caleb Taylor Caleb Taylor Marketing team List box
To move to an item press the Arrow keys.
Logan Herrera Logan Herrera Sales team
To move to an item press the Arrow keys.

Expected Behavior:
Ideally, on navigating to the list, information regarding the focused item within the list should only be read and no additional information should be read.
Also, the item count for the list item should be read.

Screenshot;
image

Issue5: Fixed
Issue6: Fixed
Issue7: Fixed
Issue8:Fixed
Issue9: Fixed.

Issue10:
Priority: Medium
Checkpoint: ACC-260.4

On adapting the screen to 320 pixels.
Press ‘F12’ and bring to 320 pixels.
Observed Behavior:
On adapting the screen to responsive design 320 pixels, content within the screen is lost.
Expected Behavior:
Al the content within the application should be accessible.

Screenshot:
image


Issue11: Fixed.
Issue12: Fixed.
Issue13: Fixed.

Issue14:Fixed.
Issue15: Fixed.
Issue16: Fixed.
Issue17: Fixed.
Issue18:Fixed.
Issue19:Fixed.
Issue20:Not reproducible

Retest environment:
-URL:https://deploy-preview-8164--fundamental-ngx.netlify.app/#/platform/approval-flow
Browser: Chrome Version 102.0.5005.115 (Official Build) (64-bit)
Screen Reader: JAWS 2022.2204.21ILM

Regards,
Karthikeyan.c

**Comment by dmitry-stepanenko** [comment]
Hi @Karthickc1994 

Regarding issues 2 and 4: I don't think it's possible to avoid reading the entire content by screen reader, it reads the entire dialog modal each time it's opened. I can add `aria-live="assertive"` for the title, but I don't think it's correct approach to go with.

**Pull request merged**